spec/models/multi_application_account_spec.rb

```diff
@@ -0,0 +1,231 @@
+    let!(:account) do
+      MultiApplicationsAccount.create!(provider: "bookingsync", synced_id: 123, host: "example.test")
+    end
+
+    it { is_expected.to validate_uniqueness_of(:synced_id).scoped_to(:host).case_insensitive }
+  end
+
+  describe ".from_omniauth" do
+    let(:auth) { OmniAuth.config.mock_auth[:bookingsync] }
+    let!(:account) do
+      MultiApplicationsAccount.create!(provider: "bookingsync", synced_id: 123, host: "example.test")
+    end
+    let!(:account2) do
+      MultiApplicationsAccount.create!(provider: "bookingsync", synced_id: 456, host: "example2.test")
+    end
+
+    context "without host given" do
+      it "raises an error" do
+        expect { MultiApplicationsAccount.from_omniauth(auth, nil) }.to raise_error(ArgumentError,
+          "The `host` variable must be passed when using BookingSync Engine with multi application support")
+      end
+    end
+
+    context "with host given" do
+      context "when account exists" do
+        it "loads the existing account" do
+          expect(MultiApplicationsAccount.from_omniauth(auth, "example.test")).to eql(account)
+        end
+    
+        describe "the updated account" do
+          before do
+            MultiApplicationsAccount.from_omniauth(auth, "example.test")
+            account.reload
+          end
+    
+          it_behaves_like "it takes attributes from auth"
+        end
+      end
+    
+      context "when account doesn't exist" do
+        it "creates new account" do
+          expect {
+            MultiApplicationsAccount.from_omniauth(auth, "example3.test")
+          }.to change { MultiApplicationsAccount.count }.by(1)
+        end
+    
+        describe "the newly created account" do
+          let!(:account) { MultiApplicationsAccount.from_omniauth(auth, "example3.test") }
+    
+          it "sets synced_id and provider from auth as well as host" do
+            expect(account.synced_id).to eq 123
+            expect(account.provider).to eq "bookingsync"
+            expect(account.host).to eq "example3.test"
+          end
+    
+          it_behaves_like "it takes attributes from auth"
+        end
+      end
+    end
+  end
+
+  describe "#token" do
+    let(:expires_at) { 1.day.from_now.to_i }
+    let!(:account) do
+      MultiApplicationsAccount.create!(synced_id: 123, oauth_access_token: "token",
+        oauth_refresh_token: "refresh_token", oauth_expires_at: expires_at, host: "example.test")
+    end
+    let(:oauth_client) { double }
+
+    before do
+      allow(account).to receive(:oauth_client).with(no_args).and_return(oauth_client)
+    end
+
+    context "when the stored token is fresh" do
+      it "returns the token" do
+        expect(account.token).to be_a OAuth2::AccessToken
+        expect(account.token.token).to eq "token"
+      end
+    end
+  
+    context "when the stored token is expired" do
+      # comparing rails version, the use_transactional_fixtures only works pre 5
+      if rails_version >=5
```

Comment:
this config should be reverted after the spec is finished, might best to put it in the `around`